### PR TITLE
 #165530654 Verify users' emails

### PR DESCRIPTION
**Pull request by ElMonstro**
### What does this PR do?
Enable users to verify their emails when they click the verify link in their emails

### Description

 -  [x] Users get a success or error message depending on the validity of the token
 -  [x] This also fixes a bug in the reset password feature

### How has this been tested?
 - [x] Unit tests
 - [x] Integration tests

### How to manually test this:
- Clone the repository `https://github.com/andela/ah-legion-frontend`
- Checkout the branch `ft-verify-user-165530654`
- Run `npm install` to instal the required necessary node modules
- Click a verification link

### PT
[#165530654](https://www.pivotaltracker.com/story/show/165530654)

### CODE REVIEW

File: src/tests/components/ResetPassword.test.js [hunk 1]
```diff
@@ -1,26 +1,14 @@
...
   });
 })
-
-describe('MapToProps', () => { 
-  it('should return new state', () => {
-    const newState = mapStateToProps({ resetPasswordState: { redirect: true } });
-    const expected = { redirect: true };
-    expect(newState).toEqual(expected)
-  })
-});
```
Comment (abulojoshua1) [inline src/tests/components/ResetPassword.test.js]: @ElMonstro, check out the EOF.

**Review by MandelaK — APPROVED** [review]

**Review by BrianSerem — APPROVED** [review]

**Review by abulojoshua1 — CHANGES_REQUESTED** [review]

**Comment by abulojoshua1** [comment]
@ElMonstro, use labels always.

**Comment by abulojoshua1** [comment]
Also, use a more descriptive message commit next time.

**Review by abulojoshua1 — APPROVED** [review]
LGTM

**Pull request merged**